C-Sharp/graphs/DFS.cs

```diff
@@ -0,0 +1,183 @@
+
+                if (endPoints.Length != 2)
+                {
+                    Console.WriteLine("Please Supply Edge is Specified Form.");
+                    return false;
+                }
+
+                if (!graph.ContainsKey(endPoints[0]) || !graph.ContainsKey(endPoints[1]))
+                {
+                    Console.WriteLine("Make Sure to Supply Valid Vertex Name For Edges");
+                    return false;
+                }
+
+                Vertex startVertex = graph[endPoints[0]], endVertex = graph[endPoints[1]];
+                Edge newEdge = new Edge() { EndVertex = endVertex };
+                startVertex.Edges.Add(newEdge);
+            }
+
+            return true;
+        }
+
+
+        /// <summary>
+        /// this method takes graph (dictionary object) as input and performs the DFS traversal
+        /// </summary>
+        /// <param name="graph">This dictionary represents graph</param>
+        static void GraphTraversal(Dictionary<string, Vertex> graph)
+        {
+            Stack<Vertex> traverseStack = new Stack<Vertex>();
+            Vertex unvisited;
+            while ((unvisited = FindUntraversedVertex(graph)) != null)
+            {
+                traverseStack.Push(unvisited);
+                while (traverseStack.Count > 0)
+                {
+                    Vertex vertex = traverseStack.Pop();
+                    vertex.Traversed = true;
+                    Console.Write($"{vertex.Value} ");
+
+                    foreach (Edge incidentEdge in vertex.Edges)
+                    {
+                        if (!incidentEdge.EndVertex.Traversed)
+                        {
+                            traverseStack.Push(incidentEdge.EndVertex);
+                        }
+                    }
+                }
+                Console.WriteLine();
+            }
+        }
+
+        /// <summary>
+        /// In case of disconnected graph, for the generation of DFS forest(instead of single tree) this method find untraversed vertices 
+        /// </summary>
+        /// <param name="graph">Dictionary representing the graph</param>
+        /// <returns></returns>
+        static Vertex FindUntraversedVertex(Dictionary<string, Vertex> graph)
+        {
+            foreach (KeyValuePair<string, Vertex> graphVertice in graph)
+            {
+                if (!graphVertice.Value.Traversed)
+                {
+                    return graphVertice.Value;
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Resetting states of all vertices to unvisited after DFS traversal
+        /// </summary>
+        /// <param name="graph"></param>
+        static void ResetGraph(Dictionary<string, Vertex> graph)
+        {
+            foreach (KeyValuePair<string, Vertex> graphVertex in graph)
+            {
+                graphVertex.Value.Traversed = false;
+            }
+        }
+
```

Comment:
Can prune these redundant lines of your code.

Reply:
Can you please elaborate what you meant by redundant lines? 
Since there is only one line of code.
Other two changes are done and will put pull request after all changes are resolved

Reply:
Yeah just keep a line break between stuff. 
I liked the code structure btw.

Reply:
Yes, I fixed the line break between code and comments.